dlog/commitment/src/commitment.rs

```diff
@@ -926,6 +906,60 @@ impl<F: Field> Utils<F> for DensePolynomial<F> {
     }
 }
 
+fn commit_helper<G: CommitmentCurve>(
+    scalars: &[Fr<G>],
+    basis: &[G],
+    is_zero: bool,
+    max: Option<usize>,
+) -> PolyComm<G> {
+    let n = basis.len();
+    let p = scalars.len();
+
+    // committing all the segments without shifting
+    let unshifted = if is_zero {
+        Vec::new()
+    } else {
+        (0..p / n + if p % n != 0 { 1 } else { 0 })
+            .map(|i| {
+                VariableBaseMSM::multi_scalar_mul(
+                    basis,
+                    &scalars[i * n..p]
+                        .iter()
+                        .map(|s| s.into_repr())
+                        .collect::<Vec<_>>(),
+                )
+                .into_affine()
```

Comment:
You might want to do a batch inversion here, much faster.

Reply:
You mean "batch normalization" right (ofc it involves the batch inversion...) -- yeah it would be better. This vector is quite small but still worth doing.